### PR TITLE
Adding usage/flags information for the kots cli

### PR DESCRIPTION
**Pull request by austinchambers**
Updated all the kots cli commands. I did my best with what information I had, but it needs technical validation for some commands (notably 'kots upload')

image

### CODE REVIEW

File: content/kots-cli/download.md [hunk 1]
```diff
@@ -5,4 +5,24 @@ title: kots download
 weight: 90120
 ---
 
-Once an application is running and has the Admin Console (kotsadm) deployed with it, you can run `kots download` to retrieve a copy of the application manifests from the cluster, and store them in a specific directory structure on your workstation.
+Retrieves a copy of the application manifests from the cluster, and store them in a specific directory structure on your workstation. Requires a running KOTS application with the Admin Console (kotsadm).
+
+### Usage
+```bash
+kubectl kots download [app-slug] [flags]
```
Comment (grantmiller) [inline content/kots-cli/download.md]: does this require [namespace] as well

File: content/kots-cli/upload.md [hunk 2]
```diff
@@ -5,4 +5,26 @@ title: kots upload
 weight: 90140
 ---
 
-If you have a copy of an application that was created with kots pull or kots download, you can upload it back to the Admin Console using the `kots upload` command.
+Upload Kubernetes manifests from the local filesystem to a cluster, creating a new version of the application that can be deployed. When you have a copy of an application that was created with `kots pull` or `kots download`, you can upload it back to the Admin Console using the `kots upload` command. 
+
+### Usage
+```bash
+kubectl kots upload [source] [flags]
```
Comment (grantmiller) [inline content/kots-cli/upload.md]: does this also require [namespace]

File: content/kots-cli/upstream.md [hunk 3]
```diff
@@ -5,10 +5,22 @@ title: kots upstream
...
 ---
 
-The `upstream` command interacts with a KOTS application that is installed in an Admin Console.
+The `kots upstream upgrade` fetches the latest version of the upstream application. It is functionality equivalent to clicking the Check For Updates in the Admin Console. 
 
-```shell
-kubectl kots upstream upgrade --namespace <namespace> <app-slug>
+### Usage
+```bash
+kubectl kots upstream upgrade [upstream-uri] [flags]
```
Comment (grantmiller) [inline content/kots-cli/upstream.md]: does this also require namespace

**Comment by grantmiller** [comment]
these are great, i do think we should have @marccampbell read through them quickly for technical validation. Also, i think there might be a few other commands that need full pages (ie. version)

**Comment by austinchambers** [comment]
@grantmiller FYI -- I copied the output from the help command, which is why [namespace] wasn't included. I can fix this here, but then we also need to fix the help command (as I know we need to do for upstream).

File: content/kots-cli/upstream.md (hunk 3 above)
Comment (austinchambers) [reply content/kots-cli/upstream.md]: namespace is included in the [flags]. Strictly speaking, not a required argument, so I left it out of the usage in this case.

Comment (austinchambers) [reply content/kots-cli/upstream.md]: (Note that the: "kubectl kots upstream --help" only returns the "help for upstream" flag).

File: content/kots-cli/upload.md (hunk 2 above)
Comment (austinchambers) [reply content/kots-cli/upload.md]: namespace is included in the [flags]. Strictly speaking, not a required argument, so I left it out of the usage in this case.

File: content/kots-cli/download.md (hunk 1 above)
Comment (austinchambers) [reply content/kots-cli/download.md]: namespace is included in the [flags]. Strictly speaking, not a required argument, so I left it out of the usage in this case. It's in the example though.

File: content/kots-cli/download.md [hunk 4]
```diff
@@ -5,4 +5,24 @@ title: kots download
...
+```
+* _Replace `[app-slug]` with the "slug" of your KOTS application_ 
+* _Provide `[flags]` according to the table below_ 
+
+| Flag                 | Type | Description |
+|:----------------------|------|-------------|
+| `--dest` | string |        the directory to store the application in (default "/Users/austin") |
+| `-h, --help`   |  |           help for download |
+| `--kubeconfig` | string |  the kubeconfig to use (default "/Users/austin/.kube/config") |
```
Comment (marccampbell) [inline content/kots-cli/download.md]: The default kubecfg flag is dynamic.  If there is a $KUBECONFIG envvar we use that, else we use $HOMEDIR/.kube/config, except on windows where we use $USERPROFILE/.kube/config.

This is the same logic that kubectl uses to determine the kubeconfig, so maybe we can just refer how that detects kubeconfig?

File: content/kots-cli/download.md [hunk 5]
```diff
@@ -5,4 +5,24 @@ title: kots download
...
+### Usage
+```bash
+kubectl kots download [app-slug] [flags]
+```
+* _Replace `[app-slug]` with the "slug" of your KOTS application_ 
+* _Provide `[flags]` according to the table below_ 
+
+| Flag                 | Type | Description |
+|:----------------------|------|-------------|
+| `--dest` | string |        the directory to store the application in (default "/Users/austin") |
```
Comment (marccampbell) [inline content/kots-cli/download.md]: This defaults to your home directory ($HOME var for Mac and Linux, $USERPROFILE dir on Windows). I think that this doc should not contain your homedir, but an indicator that this default is environment specific.

File: content/kots-cli/download.md [hunk 6]
```diff
@@ -5,4 +5,24 @@ title: kots download
...
+| `-h, --help`   |  |           help for download |
+| `--kubeconfig` | string |  the kubeconfig to use (default "/Users/austin/.kube/config") |
+| `-n, --namespace` | string |    the namespace to download from (default "default") |
+| `--overwrite` |   |       overwrite any local files, if present |
```
Comment (marccampbell) [inline content/kots-cli/download.md]: this is a `bool` type.

Comment (marccampbell) [reply content/kots-cli/download.md]: A quick glance, I think all of the missing types in this doc are bool types

File: content/kots-cli/upload.md [hunk 7]
```diff
@@ -5,4 +5,26 @@ title: kots upload
 weight: 90140
 ---
 
-If you have a copy of an application that was created with kots pull or kots download, you can upload it back to the Admin Console using the `kots upload` command.
+Upload Kubernetes manifests from the local filesystem to a cluster, creating a new version of the application that can be deployed. When you have a copy of an application that was created with `kots pull` or `kots download`, you can upload it back to the Admin Console using the `kots upload` command.
```
Comment (marccampbell) [inline content/kots-cli/upload.md]: ~to a cluster~ to the Admin Console?  I wonder if saying that it uploads to the cluster could be interpreted as a way of saying that the application is deployed?  This command just makes it available as a new version in the Admin Console.

File: content/kots-cli/reset-password.md [hunk 8]
```diff
@@ -6,3 +6,21 @@ weight: 90160
 ---
 
 If you’ve deployed an application with the admin console, the `kots reset-password` command will change the bcrypted password hash in the cluster, allowing you to log in again.
+
+### Usage
+```bash
+kubectl kots reset-password [namespace] [flags]
+```
+* _Replace `[namespace]` with the namespace where your KOTS application resides_
```
Comment (marccampbell) [inline content/kots-cli/reset-password.md]: Specifically, it's not where the application resides, but where the Admin Console is deployed

File: content/kots-cli/download.md (hunk 6 above)
Comment (austinchambers) [reply content/kots-cli/download.md]: @marccampbell For all the issues with KOTS command-line flags, these were copied from kots --help, so if we want to change it here, we should probably also change it there. 

As for the missing types, I was thinking that when unspecified, it is not followed by a value. For example, I think usage is '--overwrite' and not '--overwrite true'

File: content/kots-cli/download.md (hunk 4 above)
Comment (austinchambers) [reply content/kots-cli/download.md]: What is displayed is what is rendered by "--help". I'll go ahead and add the helpful details about the flag.

File: content/kots-cli/download.md (hunk 5 above)
Comment (austinchambers) [reply content/kots-cli/download.md]: I'll go ahead and add the helpful details about the flag. If we change it here, we may also want to change what is displayed for --help.

File: content/kots-cli/upload.md (hunk 7 above)
Comment (austinchambers) [reply content/kots-cli/upload.md]: Thanks. Will adjust wording.

Comment (austinchambers) [reply content/kots-cli/upload.md]: Note that the wording from "--help" is presently "Upload Kubernetes manifests from the local filesystem to a cluster, creating a new version of the application that can be deployed." So we probably want to update that as well.

File: content/kots-cli/reset-password.md (hunk 8 above)
Comment (austinchambers) [reply content/kots-cli/reset-password.md]: Will update wording.

File: content/kots-cli/download.md (hunk 4 above)
Comment (austinchambers) [reply content/kots-cli/download.md]: Saying default is $KUBECONFIG. If unset, then $HOME/.kube/config. Windows users generally know that $HOME = $USERPROFILE (saying this messed with the column presentation)

File: content/kots-cli/download.md (hunk 5 above)
Comment (austinchambers) [reply content/kots-cli/download.md]: Saying (defaults to $HOME).

File: content/kots-cli/download.md (hunk 6 above)
Comment (austinchambers) [reply content/kots-cli/download.md]: I'm inclined to keep this as-is, and update if we update the "--help" command.

**Pull request merged**